src/main/groovy/wooga/gradle/xcodebuild/tasks/ExportArchive.groovy

```diff
@@ -126,28 +50,26 @@ class ExportArchive extends AbstractXcodeArchiveTask implements XcodeExportActio
 
     ExportArchive() {
         super()
-        exportOptionsPlist = project.objects.fileProperty()
-        xcArchivePath = project.objects.directoryProperty()
 
-        outputPath = destinationDir.file(archiveName)
-        buildArguments = project.provider({
+        outputPath.set(destinationDir.file(archiveName))
+        setInternalArguments(project.provider({
             List<String> arguments = new ArrayList<String>()
             BuildSettings buildSettings = buildSettings.getOrElse(BuildSettings.EMPTY)
             arguments << "xcodebuild"
             arguments << "-exportArchive"
             arguments << "-exportPath" << temporaryDir.path
-            arguments << "-exportOptionsPlist" << exportOptionsPlist.get().asFile.path
-            arguments << "-archivePath" << xcArchivePath.get().asFile.path
 
-            if (additionalBuildArguments.present) {
-                additionalBuildArguments.get().each {
-                    arguments << it
-                }
+            if (exportOptionsPlist.present){
+                arguments << "-exportOptionsPlist" << exportOptionsPlist.get().asFile.path
+            }
+
+            if (xcArchivePath.present) {
+                arguments << "-archivePath" << xcArchivePath.get().asFile.path
```

Comment:
These values are not optiononal. There is no reason to check for `isPresent` since they are not marked `@Optional`